lib/floe/runner/container_runner/docker.rb

```diff
@@ -1,8 +1,8 @@
 # frozen_string_literal: true
 
 module Floe
-  class Workflow
-    class Runner
+  class Runner
+    class ContainerRunner
```

Comment:
I thought the beauty of the old interface was that all runners, regardless of the scheme had the same runner interface.
The task calls into any runner, just picking the runner based upon the scheme.

So a task would work with a bash or jq runner or a "docker" runner (yes, this has multiple possible backends. And the same asl will work with any of the backends so you can use one backend for developemant and production with the same asl file)

So I was thinking that we wouldn't need to distinguish between a Container Runner and any other runner. Changing `Floe::Workflow` to `Floe::Runner` or `Floe::Runner::Base` makes sense to me (prefer the shorter form)

Reply:
`ContainerRunner` naming here was intentional.  I was concerned naming it `DockerRunner` and that there might be confusion around the scheme vs the underlying runner choice.  That is one of the problems with our docker scheme is this particular confusion.  I am perfectly happy to change this back to DockerRunner to match the schema name.

Reply:
> I thought the beauty of the old interface was that all runners, regardless of the scheme had the same runner interface.
> The task calls into any runner, just picking the runner based upon the scheme.

Yeah I agree, I was thinking of this like the active_record connection_adapters where there is an abstract_adapter that defines the interface and each plugin implements that interface.

Reply:
I didn't really change that interface, so it still follows that pattern.  The ContainerRunner just happens to resolve the specific class dynamically based on configuration.  I guess it's very similar to the active record jdbc adapter, which talks jdbc, and part of the configuration is telling it which database to connect to through jdbc, whether it's mysql, postgresql, etc.  That latter connection type is resolved later.

Reply:
>  The ContainerRunner just happens to resolve the specific class dynamically based on configuration

Right I'm not talking about that part of the dynamic resolver, I'm thinking about Task looking up the runner from the scheme and calling the Runner.  We can take docker/podman/kubernetes out of it for this and think about an Ansible runner that has to implement an interface in order to be used by Task.

Reply:
Ignore above, I thought the runner wasn't inheriting from the base runner class anymore